charmhelpers/contrib/openstack/vaultlocker.py

```diff
@@ -37,9 +37,28 @@ def __init__(self, secret_backend=None):
         )
 
     def __call__(self):
+        import hvac
+        ctxt = {}
+        # NOTE(hopem): see https://bugs.launchpad.net/charm-helpers/+bug/1849323
         db = unitdata.kv()
+        # get last-token (back-compat)
         last_token = db.get('last-token')
+        # currently known-good secret-id
         secret_id = db.get('secret-id')
+        # log of tokens we see in this iteration
+        current_tokens = ["sentinel"]
+        # log of tokens we have already tried and, if known, the secret-id
+        # retreived.
+        used_tokens = db.get('used-token')
```

Comment:
used_tokens = db.get('used-tokens') # Missing s at the end

Reply:
ack